### PR TITLE
Tuneable Stage, Scheduler, ScheduledMutator

### PR DESCRIPTION
**Pull request by domenukk**
This allows users/fuzzers to change values at runtime.

### CODE REVIEW

**Comment by domenukk** [comment]
I'm not sure if it makes sense to have the current settings in State, it'll be slower that way.
Maybe instead I'd want a callback in which the fuzzer can decide how to react

File: libafl/src/stages/mutational.rs [hunk 1]
```diff
@@ -37,7 +37,7 @@ where
     fn mutator_mut(&mut self) -> &mut M;
 
     /// Gets the number of iterations this mutator should run for.
-    fn iterations(&self, state: &mut Z::State, corpus_idx: usize) -> Result<usize, Error>;
+    fn iterations(&self, state: &mut Z::State, corpus_idx: usize) -> Result<u64, Error>;
```
Comment (domenukk) [inline libafl/src/stages/mutational.rs:40]: until now, `iterations` in MutationalStage and StackedMutator was a different type

**Comment by domenukk** [comment]
Does this make sense or not?

**Comment by tokatoka** [comment]
I don't know what we want to achieve here
so we have a metadata
```
#[derive(Default, Clone, Copy, Eq, PartialEq, Debug, Serialize, Deserialize)]
 struct TuneableScheduledMutatorMetadata {
     next: Option<usize>,
     iters: Option<u64>,
 }
```
with this metadata used in `get_next`, does it mean that `next` is the one mutation we want to use and apply that mutation only for `iters` times (?)
I think it makes more sense if iters is a Vec<usize> instead

**Comment by domenukk** [comment]
The idea is that you can set whatever the stage should do via the metadata

**Comment by tokatoka** [comment]
I see
I mean, isn't it better to make `TuneableScheduledMutatorMetadata` to have a Vec<usize>, and then you can set multiple different mutations.
Else, it only uses 1 mutation operator at a time.

**Comment by domenukk** [comment]
Ah so instead of picking the next mutation, picking the next n mutations? maybe?

**Comment by tokatoka** [comment]
yeah that's what I mean

**Comment by domenukk** [comment]
cc @tokatoka

**Pull request merged**